### DISCUSSION TITLE
Interface to the most common graph libraries and their graphs? Or examples in the documentation?

### DISCUSSION THREAD

**Discussion by HeWeMel** [HeWeMel/nographs#16] [Proposed features that need discussions]
NoGraphs could provide functionality to adapt graphs that have been created by or with the most common graph libraries.

But should this be done? Why / why not? Alternatives?

**Comment by HeWeMel** [comment]
Pro: Example graphs of such a library could be used as example for NoGraphs

**Comment by HeWeMel** [comment]
Contra: NoGraphs is made for 'graphs that can not or should not be fully computed, stored or adapted'. The graphs of such libraries are no typical examples of the usage of NoGraphs.

**Reply by leshabirukov** [reply]
Thats depends on what exactly we doing, interaction can be various. One can use `nog` as glue between other graph formats, or, if you prefer to speak more about high level features, graph algorithms, do things like generate all paths in external stored graph, and store it back to some format.

**Reply by HeWeMel** [reply]
Good point, thanks. And in addition to the proposal 'NoGraphs adapts graphs of other libraries' above, but within the headline 'interface to ...  libraries', it leads to a second proposal:

NoGraphs could provide functionality and/or documentation to export found paths or traversed nodes/edges/subgraphs to other libraries / in their file formats.

**Comment by HeWeMel** [comment]
Contra: Library bloat

**Comment by HeWeMel** [comment]
Pro: Existing application graphs could be adapted for NoGraphs, e.g., while evaluating NoGraphs

**Comment by HeWeMel** [comment]
Alternative: A documentation section with examples for such adaptions
